src/NosCore.GameObject/Providers/NRunProvider/Handlers/WOpenHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿//  __  _  __    __   ___ __  ___ ___
+// |  \| |/__\ /' _/ / _//__\| _ \ __|
+// | | ' | \/ |`._`.| \_| \/ | v / _|
+// |_|\__|\__/ |___/ \__/\__/|_|_\___|
+// 
+// Copyright (C) 2019 - NosCore
+// 
+// NosCore is a free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using ChickenAPI.Packets.ClientPackets.Npcs;
+using ChickenAPI.Packets.Enumerations;
+using ChickenAPI.Packets.ServerPackets.UI;
+using NosCore.GameObject.ComponentEntities.Interfaces;
+using NosCore.GameObject.Networking.ClientSession;
+
+namespace NosCore.GameObject.Providers.NRunProvider.Handlers
+{
+    public class WOpenEventHandler : IEventHandler<Tuple<IAliveEntity, NrunPacket>, Tuple<IAliveEntity, NrunPacket>>
+    {
+        public bool Condition(Tuple<IAliveEntity, NrunPacket> item)
+        {
+            return item.Item2.Runner == NrunRunnerType.ProbabilityUIs && item.Item1 != null;
+        }
+
+        public void Execute(RequestData<Tuple<IAliveEntity, NrunPacket>> requestData)
+        {
+            if (requestData.ClientSession.Character.MapInstanceId != requestData.Data.Item1.MapInstanceId)
+            {
+                return;
+            }
+            if (requestData.Data.Item2.Type != null)
+            {
+                requestData.ClientSession.SendPacket(
+                    new WopenPacket
+                    {
+                        Type = (WindowType)requestData.Data.Item2.Type,
```

Comment:
Add a todo to use window type in the packet

Reply:
#843 created

Reply:
NVM it's a nrun the time depend